Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java

```diff
@@ -50,14 +50,30 @@
 import org.goobi.production.constants.FileNames;
 import org.goobi.production.constants.Parameters;
 import org.goobi.production.flow.jobs.HistoryAnalyserJob;
+import org.goobi.production.plugin.PluginLoader;
```

Comment:
Order of the imports is changed to incorrectly.

Reply:
That is a matter of taste. The Eclipse algorithm sorts by package first, then class name. `"org.goobi.production.plugin" < "org.goobi.production.plugin.CataloguePlugin"` because it is shorter (`"ab" < "abc"`). I will change it back, but it will change again on the next save. There is no setting to get to fully alphabetical sorting, ignoring the packages.

Reply:
So Eclipse is not sorting this alphabetical.

Reply:
No, first alphabetical by package, then the package members alphabetical. In general, it is more readable that way, since packages do not get mixed. But it *is not alphabetical*.

Reply:
How would you like to define such and ordering in Checkstyle config? Idea and Checkstyle use real alphabetical order.

Reply:
Personally, I would prefer to import packages rather than simple classes (using `.*` imports). To my understanding, this is what imports were designed like. The possibility to import single classes was introduced to disambiguite cases where two packages contain a class with the same name. But that’s another discussion.

I would find it sufficient to decide that everybody just sets their IDE to sort imports (to the best possible), and not check it with checkstyle. Sorting imports helps a litte bit to avoid merge conflicts (instead everybody adding new imports to the end of the list unsorted, which will always lead to merge conflicts, because there are inserts at the same line then). This is still given.

By the way, if they happen however, fixing merge conflicts on imports is easy. Just copy all imports from both files into the result and let the IDE organize them then. All duplicates or unused imports will be removed. (At least with Eclipse. I assume other IDEs do it the same way.)